protocol/v2/types/crypto.go

```diff
@@ -82,3 +127,469 @@ func VerifyReconstructedSignature(sig *bls.Sign, validatorPubKey []byte, root [3
 func rootHex(r [32]byte) string {
 	return hex.EncodeToString(r[:])
 }
+
+var Verifier = NewBatchVerifier(runtime.NumCPU(), 10, time.Millisecond*5)
+
+func init() {
+	go Verifier.Start()
+}
+
```

Comment:
I'm not a fan of this approach, it makes testing (especially integration) and replacing harder, I would want to see this being created at the top level and passed down to components that need this. also there's nothing terminating the goroutine. 
this could be taken out to other task not as part of this pr. but should be done IMO.

Reply:
I totally agree and I don't like it either. However, I took it as-is from Moshe's branch and didn't touch it because I was not sure that it will be kept because if verification performance issues. Now we decided to disable batch verification, so I deleted this.